main.py

```diff
@@ -102,51 +97,33 @@ def get_code_to_write(description, code):
 
     # Load the script
     with open(f"{FOLDER}/{automation_script}/check.py", "r") as f:
-        code = f.read()
-
-    # Chop off the final line
-    code = "\n".join(code.split("\n")[:-2])
-
-    # Show the description of the automation, which is the doc-string at the top of code
-    description = code[3:code[3:].index('"""') + 3]
-    st.text(description)
-    
-    # Exec the code and get any defined functions from it
-    functions_before = [f for f in locals().values() if callable(f)]
-    exec(code)
-    functions = [f for f in locals().values() if callable(f) and f not in functions_before]
-    new_functions = []
-    for f in functions:
-        # Filter them out if they are from mitosheet
-        if "mitosheet" not in str(inspect.getmodule(f)) :
-            new_functions.append(f)
-
-    if len(new_functions) != 1:
-        st.error('Please make sure the defined Python script has just one functiond defined in it.')       
-        st.stop()
-
-    automation_function = new_functions[0]
-    # Read in the new file to run the automation on
-    file = st.file_uploader("Upload a new file to run the automation on.", type=["csv"])
-
-    # Run the automation function on the new file
-    if file:
-        df = pd.read_csv(file)
-        automation_df = automation_function(df)
-
-        # Allow users to download this new dataframe
-
-        st.success("Finished automation.")
+        analysis = MitoAnalysis.from_json(f.read())
+
+    # Read in params
+    updated_metadata = {}
+    for param in analysis.get_param_metadata(param_type='import'):
+        new_param = st.file_uploader(param['name'])
+        if new_param and param['subtype'] == 'import_dataframe':
+            new_param = pd.read_csv(new_param)
+        if new_param is not None:
+            updated_metadata[param['name']] = new_param
+
+    run = st.button('Run')
+    if run:
+        print('running')
+        result = analysis.run(**updated_metadata)
 
         @st.cache_data
         def convert_df(df):
             # IMPORTANT: Cache the conversion to prevent computation on every rerun
             return df.to_csv().encode('utf-8')
```

Comment:
This line of code is erroring. See error here:
<img width="1640" alt="Screen Shot 2023-10-19 at 9 03 39 PM" src="https://github.com/mito-ds/mito-for-streamlit-recon/assets/21002431/6b387338-3f66-41c6-8a7a-3406f081d302">

Reply:
Oops fixed. This was because I was assuming the result was a list but it was actually sometimes just a DF.